modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/auth/SecondaryStoreUserLoginTestCase.java

```diff
@@ -0,0 +1,269 @@
+            ".app/oauth2client";
+    public static final String targetApplicationUrl = "http://localhost:" + TOMCAT_PORT + "%s";
+    public static final String appUserAuthorizePath = "/playground2/oauth2-authorize-user.jsp";
+    private static final Log log = LogFactory.getLog(OIDCAbstractIntegrationTest.class);
+    private static final Log LOG = LogFactory.getLog(TomcatInitializerTestCase.class);
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        super.init();
+        CookieStore cookieStore = new BasicCookieStore();
+
+//      Register a secondary user store
+        userStoreConfigAdminServiceClient =
+                new UserStoreConfigAdminServiceClient(backendURL, sessionCookie);
+        userMgtClient = new UserManagementClient(backendURL, getSessionCookie());
+        try {
+            UserStoreDTO userStoreDTO = userStoreConfigAdminServiceClient.createUserStoreDTO(JDBC_CLASS, DOMAIN_ID,
+                    userStoreConfigUtils.getJDBCUserStoreProperties(USER_STORE_DB_NAME));
+            userStoreConfigAdminServiceClient.addUserStore(userStoreDTO);
+            Thread.sleep(5000);
+        } catch (Exception e) {
+            log.error(e);
+        }
+        boolean isSecondaryUserStoreDeployed = userStoreConfigUtils.waitForUserStoreDeployment(
+                userStoreConfigAdminServiceClient, DOMAIN_ID);
+
+        if (isSecondaryUserStoreDeployed) {
+
+//          Creating users in the primary and secondary user stores
+            try {
+                addUserIntoJDBCUserStore(primUsername, primPassword, false);
+                addUserIntoJDBCUserStore(secUsername, secPassword, true);
+            } catch (Exception e) {
+                log.error(e);
+            }
+
+//          Creating and starting application on tomcat
+            initAndCreatePlaygroundApplication();
+            startTomcat();
+            client = HttpClientBuilder.create().setDefaultCookieStore(cookieStore).build();
+
+//          To generate sessionDataKey
+            try {
+                testSendAuthenticationRequest(playgroundApp, client);
+            } catch (Exception e) {
+                log.error(e);
+            }
+        } else {
+            log.error("Secondary user store is not deployed");
+        }
+    }
+
+    @DataProvider(name = "userCredentialProvider")
+    public static Object[][] userCredentialProvider() {
+
+        return new Object[][]{
+                {primUsername, primPassword},
+                {secUsername, secPassword},
+        };
+    }
+
+    @Test(groups = "wso2.is", description = "Check the secondary user store user login flow",
+            dataProvider = "userCredentialProvider")
+    public void testUserLogin(String username, String password) {
+
+        try {
+            testAuthentication(playgroundApp, username, password);
+        } catch (Exception e) {
+            log.error("Error: " + e);
```

Comment:
If there is a error, that means test is failed, right? If so you should throw it and mark the test has failed.